packages/webpack-cli/lib/plugins/WebpackCLIPlugin.js

```diff
@@ -18,11 +19,13 @@ class WebpackCLIPlugin {
                 let progressPluginExists;
 
                 if (compiler.options.plugins) {
-                    progressPluginExists = Boolean(compiler.options.plugins.find((e) => e instanceof ProgressPlugin));
+                    progressPluginExists = Boolean(compiler.options.plugins.find((plugin) => plugin instanceof ProgressPlugin));
                 }
 
                 if (!progressPluginExists) {
-                    new ProgressPlugin().apply(compiler);
+                    const isProfile = this.options.progress === 'profile';
+
+                    new ProgressPlugin({ profile: isProfile }).apply(compiler);
```

Comment:
What if an invalid value is provided `webpack --progress=invalid`. Should we throw error ?

Reply:
Fixed, please review again